pkg/controller/migcluster/validation.go

```diff
@@ -216,6 +226,50 @@ func (r ReconcileMigCluster) testConnection(cluster *migapi.MigCluster) error {
 	return nil
 }
 
+// Validate the Exposed registry route
+func (r ReconcileMigCluster) validateRegistryRoute(cluster *migapi.MigCluster) error {
+	if cluster.Spec.IsHostCluster {
+		return nil
+	}
+
+	if cluster.Status.HasCriticalCondition() {
+		return nil
+	}
+
+	if cluster.Spec.ExposedRegistryPath != "" {
+		url := "https://" + cluster.Spec.ExposedRegistryPath + "/v2/"
+		restConfig, err := cluster.BuildRestConfig(r.Client)
+		token := restConfig.BearerToken
+
+		tr := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+		client := &http.Client{Transport: tr}
+
+		req, err := http.NewRequest("GET", url, nil)
+
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+
+		req.Header.Set("Authorization", "bearer "+token)
+
+		res, err := client.Do(req)
+		if err != nil {
+			cluster.Status.SetCondition(migapi.Condition{
+				Type:     InvalidRegistryRoute,
+				Status:   True,
+				Reason:   RouteTestFailed,
+				Category: Critical,
```

Comment:
I noticed this is a critical condition. We don't always require a registry route to perform migrations. 

I guess the idea here is that _if_ a user supplies a value then we will block MigCluster from being ready (and therefore block all associated migrations), but they can remove `Spec.ExposedRegistryPath` if they can't figure out the registry route and want to do an indirect migration.

Reply:
We don't require a registry route, which is why this PR doesn't fail validation of the MigCluster if the registry route is absent. I think it's reasonable to expect that *if* a user provides a route that it be valid, and if it's not valid, make the cluster non-ready. If the user is doing indirect image migration, the remedy for this critical condition would be to clear the registry route entry on the MigCluster.